datadog/fwprovider/resource_datadog_integration_aws_account.go

```diff
@@ -0,0 +1,902 @@
+}
+
+type tracesConfigModel struct {
+	XrayServices *xrayServicesModel `tfsdk:"xray_services"`
+}
+
+type xrayServicesModel struct {
+	IncludeAll  types.Bool `tfsdk:"include_all"`
+	IncludeOnly types.List `tfsdk:"include_only"`
+}
+
+func NewIntegrationAwsAccountResource() resource.Resource {
+	return &integrationAwsAccountResource{}
+}
+
+func (r *integrationAwsAccountResource) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	providerData, _ := request.ProviderData.(*FrameworkProvider)
+	r.Api = providerData.DatadogApiInstances.GetAWSIntegrationApiV2()
+	r.Auth = providerData.Auth
+}
+
+func (r *integrationAwsAccountResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "integration_aws_account"
+}
+
+func (r *integrationAwsAccountResource) ConfigValidators(ctx context.Context) []resource.ConfigValidator {
+	return []resource.ConfigValidator{
+		resourcevalidator.ExactlyOneOf(
+			awsRegionsPath,
+		),
+		resourcevalidator.Conflicting(
+			awsRegionsPath.AtName("include_all"),
+			awsRegionsPath.AtName("include_only"),
+		),
+		resourcevalidator.Conflicting(
+			authConfigKeysPath,
+			authConfigRolePath,
+		),
+		resourcevalidator.ExactlyOneOf(
+			authConfigKeysPath,
+			authConfigRolePath,
+		),
+		resourcevalidator.RequiredTogether(
+			authConfigRolePath,
+			authConfigRolePath.AtName("role_name"),
+		),
+		resourcevalidator.RequiredTogether(
+			authConfigKeysPath,
+			authConfigKeysPath.AtName("access_key_id"),
+			authConfigKeysPath.AtName("secret_access_key"),
+		),
+		resourcevalidator.ExactlyOneOf(
+			lambdaForwarderPath,
+		),
+		resourcevalidator.ExactlyOneOf(
+			namespaceFiltersPath,
+		),
+		resourcevalidator.Conflicting(
+			namespaceFiltersPath.AtName("include_only"),
+			namespaceFiltersPath.AtName("exclude_only"),
+		),
+		resourcevalidator.ExactlyOneOf(
+			xrayServicesConfigPath,
+		),
+		resourcevalidator.Conflicting(
+			xrayServicesConfigPath.AtName("include_all"),
+			xrayServicesConfigPath.AtName("include_only"),
+		),
+	}
+}
+
+func (r *integrationAwsAccountResource) ModifyPlan(ctx context.Context, request resource.ModifyPlanRequest, response *resource.ModifyPlanResponse) {
+	// Remove `metrics_config.aws_namespace_filers.exclude_only` default if `include_only` is set.
+	fwutils.RemoveDefaultIfConflictingSet(ctx, request, response, namespaceFiltersPath.AtName("exclude_only"), namespaceFiltersPath.AtName("include_only"))
+	// Remove `aws_regions.include_all` default if `include_only` is set.
+	fwutils.RemoveDefaultIfConflictingSet(ctx, request, response, awsRegionsPath.AtName("include_all"), awsRegionsPath.AtName("include_only"))
+	// Remove `traces_config.xray_services.include_only` default if `include_all` is set.
+	fwutils.RemoveDefaultIfConflictingSet(ctx, request, response, xrayServicesConfigPath.AtName("include_only"), xrayServicesConfigPath.AtName("include_all"))
```

Comment:
How is this ModifyPlan used? I'd prefer just failing in these cases rather than trying to modify the plan. It's really not clear which field the customer meant to set if both are set imo.

Reply:
A validation error is returned if both are set in the user's config